src/batch_eval.rs

```diff
@@ -0,0 +1,79 @@
+//! `kurobako run` command.
+use crate::problem::KurobakoProblemRecipe;
+use crate::solver::KurobakoSolverRecipe;
+use kurobako_core::epi::channel::{MessageReceiver, MessageSender};
+use kurobako_core::json;
+use kurobako_core::problem::ProblemRecipe as _;
+use kurobako_core::problem::{Evaluator as _, Problem as _, ProblemFactory as _};
+use kurobako_core::registry::FactoryRegistry;
+use kurobako_core::rng::ArcRng;
+use kurobako_core::trial::{Params, Values};
+use kurobako_core::{ErrorKind, Result};
+use serde::Deserialize;
+use serde::Serialize;
+use std::io;
+use structopt::StructOpt;
+
+/// Options of the `kurobako evaluate` command.
+#[derive(Debug, Clone, StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct BatchEvaluateOpt {
+    /// Evaluation target problem.
+    #[structopt(long, parse(try_from_str = json::parse_json))]
+    pub problem: KurobakoProblemRecipe,
+
+    /// Random seed.
+    #[structopt(long)]
+    pub seed: Option<u64>,
+}
+/// Messages that are used to communicate with external solvers.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[allow(missing_docs)]
+#[serde(tag = "type", rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum BatchEvaluateMessage {
+    EvalCall {
+        /// Parameters to be evaluated.
+        params: Params,
+        step: Option<u64>,
+    },
+    EvalReply {
+        values: Values,
+    },
+    EvalEnd,
+}
+
+impl BatchEvaluateOpt {
+    /// Evaluates the given parameters.
+    pub fn run(&self) -> Result<()> {
+        let mut rx: MessageReceiver<BatchEvaluateMessage, _> = MessageReceiver::new(io::stdin());
```

Comment:
I think that we can use `kurobako_core::json::load()` function (https://github.com/optuna/kurobako/blob/master/kurobako_core/src/json.rs#L19) to read parameters from the stdin until it reaches EOS. 
IMO, the load function is simpler than `MessageReceiver` for this purpose and, by using the load function, we could remove `BatchEvaluateMessage` enum as it would be sufficient to define a minimum struct like the following:
```rust
#[derive(Deug, Deserialize)]
struct EvalCall {
    params: Params,
    step: Option<u64>
}

impl BatchEvaluateOpt {
    pub fn run(&self) -> Result<()> {
        let calls: Vec<EvalCall> = track!(kurobako_core::json::load(io::stdin()))?;
        ...
    }
}
```